### PR TITLE
Adds fetching work details to Orcid client.

### LINKED ISSUE #1290: Add retrieving work details to Orcid client
For mapping to a publication, the full work record is needed, not just the work summary.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #1290

## Why was this change made?
It will be used to populate the publication.

## How was this change tested?
Unit

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

File: lib/orcid/client.rb [hunk 1]
```diff
@@ -8,13 +8,26 @@ class Client
...
+
+    def get(url)
+      response = conn.get(url)
       raise "ORCID.org API returned #{response.status}" if response.status != 200
 
       JSON.parse(response.body).with_indifferent_access
     end
 
-    private
+    def base_orcidid(orcidid)
```
Comment (peetucket) [inline lib/orcid/client.rb:28]: we are going to need this kind of method when adding the ORCID-based query to the WoS client (was just taking a peek at that now) ... is there a better place to put it to make it easier to use in a different class?

Comment (justinlittman) [reply lib/orcid/client.rb:28]: In general, an ORCID ID is considered the entire URL: `https://orcid.org/0000-0003-3437-349X`, not just `0000-0003-3437-349X`. That being said, I have no idea what WOS accepts.

Comment (peetucket) [reply lib/orcid/client.rb:28]: WoS wants the base identifier as well without the prefix

Comment (justinlittman) [reply lib/orcid/client.rb:28]: Good catch. Extracting.

File: lib/orcid/client.rb [hunk 2]
```diff
@@ -41,8 +43,11 @@ def add_work(orcidid, work, token)
 
     private
 
-    def base_orcidid(orcidid)
-      orcidid[-19, 19]
+    def get(url)
+      response = conn.get(url)
+      raise "ORCID.org API returned #{response.status}" if response.status != 200
```
Comment (peetucket) [inline lib/orcid/client.rb:48]: just double-checking ... will this give us an HB alert?

**Review by peetucket — APPROVED** [review]
Just have one question, otherwise looks good.

File: lib/orcid/client.rb (hunk 2 above)
Comment (justinlittman) [reply lib/orcid/client.rb:48]: If not caught and used in an environment in which HB is configured, I'd expect so.

**Pull request merged**